cmake/UseDoxygen.cmake

```diff
@@ -0,0 +1,96 @@
+# Use FindDoxygen.cmake to generate documentation.
+
+macro(UseDoxygen)
+  if (${CMAKE_VERSION} VERSION_LESS "3.9")
+    # Old versions of cmake have poor support for Doxygen generation.
+    message(WARNING "Doxygen generation only enabled for cmake 3.9 and higher")
+  else()
+    find_package(Doxygen)
+    if (DOXYGEN_FOUND)
```

Comment:
Hm, I don't think that auto creating doxygen documentation is a good idea, since 99.99% of those who runs build libevent do not need it.

So I can you make doxygen under `EVENT__DOXYGEN` and fail the target if doxygen is not found? (and some analog for autotools)

Reply:
Thanks for your advice. I added the `EVENT__DOXYGEN` option to cmake and the `enable-doxygen-doc` option to autotools. These options default to false.
cmake:
```
cmake -DEVENT__DOXYGEN=ON ..
```
autotools:
```
./configure --enable-doxygen-doc
```